source/common/network/tcp_listener_impl.cc

```diff
@@ -60,7 +60,8 @@ void TcpListenerImpl::onSocketEvent(short flags) {
       io_handle->close();
       cb_.onReject(TcpListenerCallbacks::RejectCause::GlobalCxLimit);
       continue;
-    } else if (random_.bernoulli(reject_fraction_)) {
+    } else if ((listener_accept_ != nullptr && listener_accept_->shouldShedLoad()) ||
```

Comment:
<s>I think a stat here will help with monitoring this. Not sure if the stats should just be aggregated with the Overload-Manager stats, or kept separate.</s>
Looked at the integration test, and saw that there are gauges.

Reply:
sg, take this is resolved.